### DISCUSSION TITLE
How to configure Nuxt to use .js instead of .mjs extensions?

### DISCUSSION THREAD

**Discussion by ffxsam** [nuxt/nuxt#17100] [Questions] [answered]
I'm trying to set up a Nuxt app within a [Serverless Stack](https://serverless-stack.com/) project. When defining Lambda functions in SST, it requires this format for identifying the handler code:

```js
new sst.Function(this, 'EntryPointFunc', {
  handler: 'nuxt/.output/server/index.handler',
```

The problem is, it's looking for `index.js`, not `index.mjs`. Is there some way I can configure Nuxt to use `.js` instead?

**Comment by danielroe** [comment, marked as answer]
Yes, it's possible to configure Nitro to do this. It's more complicated though, depending on whether SST supports esm. We might be able to add a cjs stub using `.js` if not (. I've opened https://github.com/unjs/nitro/issues/133 to track.

**Comment by ffxsam** [comment]
I'm actually working on a minimal SST + Nuxt repo that people can use as a template. Happy to share once it's up! The nice thing with this setup is that Nuxt won't have to make API calls to a back end, because it's _part_ of the back end.

**Comment by Eliophot** [comment]
Hello @ffxsam, have you successfully configured Nitro to generate js instead mjs files ?

**Comment by danielroe** [comment]
I've opened https://github.com/nuxt/framework/pull/6505 to allow this to be done more easily (see PR description).

But for now you can implement like this:

```js
nuxt.hook('vite:extendConfig', (config, { isClient }) {
  if (isClient) {
    // rc6
    config.build.rollupOptions.output.chunkFileNames = '[hash].mjs'
    // rc7
    config.build.rollupOptions.output.chunkFileNames = '_nuxt/[hash].mjs'
  }
})
```
